src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -178,7 +190,9 @@ bool Series::validateSizes() const
 {
     if (maxInjectionModulation.size() != HOURS_PER_YEAR
         || maxWithdrawalModulation.size() != HOURS_PER_YEAR || inflows.size() != HOURS_PER_YEAR
-        || lowerRuleCurve.size() != HOURS_PER_YEAR || upperRuleCurve.size() != HOURS_PER_YEAR)
+        || lowerRuleCurve.size() != HOURS_PER_YEAR || upperRuleCurve.size() != HOURS_PER_YEAR
+        || costInjection.size() != HOURS_PER_YEAR || costWithdrawal.size() != HOURS_PER_YEAR
+        || costLevel.size() != HOURS_PER_YEAR)
```

Comment:
Make a function and call it for each series so the user gets a more specific warning (wrong size for series XXX) instead of a generic one like here.

A lambda function could make the job